### PR TITLE
feature/assume free

### PR DESCRIPTION
**Pull request by zkronos73**
### assumeFree feauture ###
- F_MLOAD, F_HASHXn
- sugar syntax:   
  - $ + 3 => A :F_MLOAD(myvar)  ====>  myvar + 3 => A
  - B + 2  => A  :MSTORE(myvar)  ====> B + 2 => A,myvar
  - $ + 3 => A :F_MLOAD(myvar+E) ====>  myvar[E] + 3 => A
  - B + 2  => A  :MSTORE(myvar+E)  ====> B + 2 => A,myvar[E]
- indirection aligned with pil: n1 * E + n2 * RR (now the addRel could be shared if it's the same)
- HASHX(addr) => HASHX(hashAddr) (hashAddr = E, E+n, E-n). This allows uses HASH and MLOAD,MSTORE in same row

### new tests ###
Added new tests to validate output produced by zkasm

### zkasm code structure ###
Internally pass function to methods, and context (ctx) as this.

### tool compare_rom_json ###
Tool to compare two results of compilation, first with current version and second with previous version. The ideas is detect errors introduced during this version. At end this tool detect two previous errors.

### CODE REVIEW

**Review by krlosMata — APPROVED** [review]
:smile_cat:

File: src/zkasm_parser.jison [hunk 1]
```diff
@@ -28,25 +28,38 @@ ROTL_C                  { return 'ROTL_C'; }
 HASHPOS                 { return 'HASHPOS'; }
 RID                     { return 'RID'; }
 MLOAD                   { return 'MLOAD' }
+F_MLOAD                 { return 'F_MLOAD' }
 MSTORE                  { return 'MSTORE' }
+MSTORE                  { return 'F_MSTORE' }
```
Comment (hecmas) [inline src/zkasm_parser.jison]: This should be F_MSTORE

**Review by hecmas — CHANGES_REQUESTED** [review]
Fix the typo

File: src/zkasm_parser.jison (hunk 1 above)
Comment (zkronos73) [reply src/zkasm_parser.jison]: It a typo, but need to check test, because with this mistake, a F_MSTORE generate an error because not recognized.

Comment (zkronos73) [reply src/zkasm_parser.jison]: remove F_MSTORE, no sense

**Review by hecmas — APPROVED** [review]

**Pull request merged**